typer/main.py

```diff
@@ -651,7 +651,7 @@ def generate_list_convertor(
     convertor: Optional[Callable[[Any], Any]], default_value: Optional[Any]
 ) -> Callable[[Sequence[Any]], Optional[List[Any]]]:
     def internal_convertor(value: Sequence[Any]) -> Optional[List[Any]]:
-        if default_value is None and len(value) == 0:
+        if default_value is None and (value is None or len(value) == 0):
```

Comment:
If `value` can be `None`, then the type of `value` should also be updated to `Optional[Sequence[Any]]`.

Reply:
Sure. See you've already made the change. Thanks

Reply:
Yep - and making this change uncovered one more related issue that I'm fixing now.